OhmGraphite/MetricTimer.cs

```diff
@@ -106,28 +109,38 @@ private static Sensor Normalize(Sensor sensor)
             return new Sensor(identifier, name, sensor.Value);
         }
 
-        private static IEnumerable<Sensor> ReadSensors(IComputer computer)
+        private static IEnumerable<IHardware> ReadHardware(IComputer computer)
         {
             foreach (var hardware in computer.Hardware)
             {
-                hardware.Update();
-                foreach (var sensor in hardware.Sensors)
+                yield return hardware;
+
+                foreach (var subHardware in hardware.SubHardware)
                 {
-                    var id = sensor.Identifier.ToString();
+                    yield return subHardware;
+                }
+            }
+        }
 
-                    // Only report a value if the sensor was able to get a value
-                    // as 0 is different than "didn't read". For example, are the
-                    // fans really spinning at 0 RPM or was the value not read.
-                    if (sensor.Value.HasValue)
-                    {
-                        yield return new Sensor(id, sensor.Name, sensor.Value.Value);
-                    }
-                    else
-                    {
-                        Logger.Warn($"{id} did not have a value");
-                    }
+        private static IEnumerable<Sensor> ReadSensors(IHardware hardware)
+        {
+            hardware.Update();
+            foreach (var sensor in hardware.Sensors)
+            {
+                var id = sensor.Identifier.ToString();
+
+                // Only report a value if the sensor was able to get a value
+                // as 0 is different than "didn't read". For example, are the
+                // fans really spinning at 0 RPM or was the value not read.
+                if (sensor.Value.HasValue)
+                {
+                    yield return new Sensor(id, sensor.Name, sensor.Value.Value);
+                }
+                else
+                {
+                    Logger.Warn($"{id} did not have a value");
```

Comment:
For you, is this line tripped for any of the sub-sensors (ie. the fan controllers)?

If so, you or I could remove this line (or change it to Debug or similar) to avoid logspam.

Reply:
Yeah it does log each time if the fan controllers are null. Good idea, I'll set it to Debug.